### PR TITLE
enable cl telemetry  in android build and add logging, configuration, pstore probe support

### PR DESCRIPTION
**Pull request by btian1**
Add build support for cl telemetry in android system.
Add logging support for cl telemetry in android system.
Add configuration support for cl telemetry in android system.
Add support for pstore probe under android run time.

Change-Id: Id14599df4b19a34410409e0b32ee032910435e25
Signed-off-by: Tian Baofeng <baofeng.tian@intel.com>
Signed-off-by: chenxia1 <xiao.x.chen@intel.com>

### CODE REVIEW

File: android/config.h [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+#define PACKAGE "telemetrics-client"
+
+/* Define to the address where bug reports for this package should be sent. */
+#define PACKAGE_BUGREPORT "https://clearlinux.org/"
+
+/* Define to the full name of this package. */
+#define PACKAGE_NAME "telemetrics-client"
+
+/* Define to the full name and version of this package. */
+#define PACKAGE_STRING "telemetrics-client 2.2.1"
```
Comment (jurobystricky) [inline android/config.h]: This is wrong. That is why we use autoconf. The 2.2.1 is gguaranteed to be changed at some point

File: android/config.h [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+#define TM_LOG_STDERR 1
+
+/* Log through syslog apis */
+/* #undef TM_LOG_SYSLOG */
+
+/* Log through systemd apis */
+/* #undef TM_LOG_SYSTEMD */
+
+/* Version number of package */
+#define VERSION "2.2.1"
```
Comment (jurobystricky) [inline android/config.h]: Again, you cannot hardcode a version number.

File: android/config.h [hunk 3]
```diff
@@ -0,0 +1,222 @@
+#ifndef __CONFIG_H__
```
Comment (jurobystricky) [inline android/config.h]: This file should be generated by autotools.

File: src/probes/pstore_probe.c [hunk 4]
```diff
@@ -230,7 +264,13 @@ int main(int argc, char **argv)
                 // Look for only dmesg logs. Ignore other types of pstore dumps for now.
                 if (sscanf(entry->d_name, "dmesg-efi-%" PRIu64, &id) == 1) {
                         parse_id(id, &count, &part);
-                        telem_debug("DEBUG: Extracted count :%d, part : %d\n", count, part);
+                        telem_debug("dmesg-efi Extracted count :%d, part : %d\n",
+                                    count, part);
+                } else if (sscanf(entry->d_name, "dmesg-ramoops-%" PRIu64, &id) == 1) {
+                        count = id + 1;
```
Comment (jurobystricky) [inline src/probes/pstore_probe.c]: I get this warning:
./../src/probes/pstore_probe.c:270:33: warning: conversion from ‘uint64_t’ {aka ‘long unsigned int’} to ‘int’ may change value [-Wconversion]
  270 |                         count = id + 1;

Comment (btian1) [reply src/probes/pstore_probe.c]: accept

**Pull request closed without merging**